bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/config/TomlProviderTest.java

```diff
@@ -800,4 +801,62 @@ public void testTomlProviderUnionTables() {
             }
         }
     }
+
+    @Test
+    public void testAdditionalFieldsInOpenRecords() {
+        RecordType recordType = TypeCreator.createRecordType("Person", ROOT_MODULE, SymbolFlags.READONLY,
+                new HashMap<>(), TYPE_ANYDATA, false, 6);
+        VariableKey recordVar = new VariableKey(ROOT_MODULE, "recordVar", recordType, true);
+        RuntimeDiagnosticLog diagnosticLog = new RuntimeDiagnosticLog();
+        ConfigResolver configResolver = new ConfigResolver(Map.ofEntries(Map.entry(ROOT_MODULE,
+                new VariableKey[]{recordVar})), diagnosticLog, List.of(new TomlFileProvider(ROOT_MODULE, getConfigPath(
+                "OpenRecordConfig.toml"), Set.of(ROOT_MODULE))));
+        Map<VariableKey, Object> configValueMap = configResolver.resolveConfigs();
+        Assert.assertEquals(diagnosticLog.getErrorCount(), 0);
+        Assert.assertEquals(diagnosticLog.getWarningCount(), 0);
+        Object recordValue = configValueMap.get(recordVar);
+        Assert.assertTrue(recordValue instanceof BMap<?, ?>, "Non-map value received");
+        BMap<?, ?> record = (BMap<?, ?>) recordValue;
+        Assert.assertEquals(record.getIntValue(fromString("intValue")), (Long) 25L);
+        Assert.assertEquals(record.getFloatValue(fromString("floatValue")), 25.54);
+        Assert.assertEquals(record.getStringValue(fromString("stringValue")), fromString("test"));
+        Assert.assertEquals(record.getBooleanValue(fromString("booleanValue")), (Boolean) true);
+    }
```

Comment:
If we can move the test initialization and other parts that are not directly relevant to the main test to a different method or other place, it would add clarity to the test? WDYT? This is because a test has to be easy to read and write as we cannot write another test to validate the current test 😀